crates/bindings-csharp/BSATN.Codegen/Type.cs

```diff
@@ -8,11 +8,23 @@ namespace SpacetimeDB.Codegen;
 
 public record MemberDeclaration(string Name, string Type, string TypeInfo)
 {
-    public MemberDeclaration(string name, ITypeSymbol type)
-        : this(name, SymbolToName(type), GetTypeInfo(type)) { }
+    public MemberDeclaration(ISymbol member, ITypeSymbol type, DiagReporter diag)
+        : this(member.Name, SymbolToName(type), "")
+    {
+        try
+        {
+            TypeInfo = GetTypeInfo(type);
+        }
+        catch (Exception e)
```

Comment:
Is that wide a catch type needed? Won't that turn an internal error into an unsupported type?

Reply:
Yeah but an internal error during type resolution is due to an unsupported type one way or another, and the detailed error message will contain the original exception text anyway.

So even if there's an internal error, this will at least limit the context to one specific field and the text will explain why it couldn't be handled.